src/main/java/org/htwkvisu/scoring/ScoringCalculator.java

```diff
@@ -0,0 +1,168 @@
+package org.htwkvisu.scoring;
+
+import javafx.geometry.Point2D;
+
+import java.util.*;
+
+/**
+ * ScoringCalculator calculates a scoring values of a given point.
+ * Every category can get a specific weight value for the relative importance in the score value.
+ */
+public class ScoringCalculator {
+
+    public static final String CATEGORY_HEALTH = "Health";
+    public static final String CATEGORY_INFRASTRUCTURE = "Infrastructure";
+    public static final String CATEGORY_EDUCATION = "Education";
+
+    private Set<String> categories = new HashSet<>();
+    private Map<String, Double> weights = new HashMap<>();
+    private List<IScorable> pois = new LinkedList<>();
+
+    public ScoringCalculator() {
+
+    }
+
+    /**
+     * Add new POI (IScorable).
+     *
+     * @param poi POI
+     */
+    public void addPOI(IScorable poi) {
+        pois.add(poi);
+    }
+
+    /**
+     * Add category.
+     *
+     * @param cat    Category
+     * @param weight Weight
+     */
+    public void addCategory(String cat, double weight) {
+        categories.add(cat);
+        weights.put(cat, weight);
+    }
+
+    /**
+     * Set weight of specific category.
+     *
+     * @param cat Category
+     * @param weight Weight value
+     */
+    public void setCategoryWeight(String cat, double weight) {
+        if (categories.contains(cat) && weights.containsKey(cat)) {
+            weights.put(cat, weight);
+        } else {
+            throw new IllegalArgumentException("Category not known!");
+        }
+    }
+
+    /**
+     * Get weight of specific category.
+     *
+     * @param cat Category
+     * @return Weight value
+     */
+    public double getCategoryWeight(String cat) {
+        if (!weights.containsKey(cat)) {
+            throw new IllegalArgumentException("Weight not known!");
+        }
+
+        return weights.get(cat);
+    }
+
+    /**
+     * Remove specific category.
+     *
+     * @param cat Category
+     */
+    public void removeCategory(String cat) {
+        if (!categories.contains(cat) || !weights.containsKey(cat)) {
+            throw new IllegalArgumentException("Category not known!");
+        }
+
+        categories.remove(cat);
+        weights.remove(cat);
+    }
+
+    /**
+     * Removes all categories
+     */
+    public void resetCategories() {
+        categories.clear();
+        weights.clear();
+    }
+
+    /**
+     * Get all categories as list.
+     *
+     * @return List of category identifiers
+     */
+    public List<String> getCategories() {
+        return new LinkedList<>(categories);
+    }
+
+    /**
+     * Get all POIs as list.
+     *
+     * @return List of POIs
+     */
+    public List<IScorable> getPOIs() {
```

Comment:
please remove the javadoc of getters and setters, because the behaviour of all of them is known